src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ExceptionsPipelineTests.cs

```diff
@@ -213,14 +214,194 @@ public Task EventExceptionsPipeline_ArrayException()
                 {
                     TestExceptionsStore.ExceptionInstance instance = Assert.Single(instances);
                     Assert.NotNull(instance);
-                    Assert.NotEqual(0UL, instance.GroupId);
+                    Assert.NotEqual(0UL, instance.Id);
                     Assert.Equal(typeof(IndexOutOfRangeException).FullName, instance.TypeName);
                     Assert.False(string.IsNullOrEmpty(instance.Message));
 
                     ValidateStack(instance, "ThrowIndexOutOfRangeException", "System.Private.CoreLib.dll", "System.ThrowHelper");
                 });
         }
 
+        /// <summary>
+        /// Tests that exceptions with inner exceptions that haven't been thrown are detectable.
+        /// </summary>
+        [Fact]
+        public Task EventExceptionsPipeline_InnerUnthrownException()
+        {
+            const int ExpectedInstanceCount = 2;
+
+            return Execute(
+                TestAppScenarios.Exceptions.SubScenarios.InnerUnthrownException,
+                ExpectedInstanceCount,
+                validate: instances =>
+                {
+                    Assert.Equal(ExpectedInstanceCount, instances.Count());
+
+                    TestExceptionsStore.ExceptionInstance instance1 = instances.First();
```

Comment:
nit: consider renaming `instance1` and `instance2` to make the outer/inner relationship more obvious at a glance